parser/src/main/scala/hmda/parser/fi/lar/LarCsvParser.scala

```diff
@@ -2,144 +2,154 @@ package hmda.parser.fi.lar
+            loanAmount
+          )
 
-      val applicant =
-        Applicant(
-          appEthnicity,
-          coAppEthnicity,
-          appRace1,
-          appRace2,
-          appRace3,
-          appRace4,
-          appRace5,
-          coAppRace1,
-          coAppRace2,
-          coAppRace3,
-          coAppRace4,
-          coAppRace5,
-          appSex,
-          coAppSex,
-          appIncome
-        )
-      val denial = Denial(denial1, denial2, denial3)
+        val geography = Geography(msa, state, county, tract)
 
-      Right(LoanApplicationRegister(
-        id,
-        respId,
-        agencyCode,
-        loan,
-        preapprovals,
-        actionType,
-        actionDate,
-        geography,
-        applicant,
-        purchaserType,
-        denial,
-        rateSpread,
-        hoepaStatus,
-        lienStatus
-      ))
-    } else {
-      Left(larErrors)
+        val applicant =
+          Applicant(
+            appEthnicity,
+            coAppEthnicity,
+            appRace1,
+            appRace2,
+            appRace3,
+            appRace4,
+            appRace5,
+            coAppRace1,
+            coAppRace2,
+            coAppRace3,
+            coAppRace4,
+            coAppRace5,
+            appSex,
+            coAppSex,
+            appIncome
+          )
+        val denial = Denial(denial1, denial2, denial3)
+
+        Right(
+          LoanApplicationRegister(
+            id,
+            respId,
+            agencyCode,
+            loan,
+            preapprovals,
+            actionType,
+            actionDate,
+            geography,
+            applicant,
+            purchaserType,
+            denial,
+            rateSpread,
+            hoepaStatus,
+            lienStatus
+          )
+        )
+      }
+      case scalaz.Failure(errors) => {
+        Left(errors.toList)
+      }
     }
   }
 
-  def checkLar(s: String): List[String] = {
-    val values = s.split('|').map(_.trim)
-    var errors = List[String]()
-    val numericFields = Map(
-      0 -> "Record Identifier",
-      2 -> "Agency Code",
-      5 -> "Loan Type",
-      6 -> "Property Type",
-      7 -> "Loan Purpose",
-      8 -> "Owner Occupancy",
-      9 -> "Loan Amount",
-      10 -> "Preapprovals",
-      11 -> "Type of Action Taken",
-      12 -> "Date of Action",
-      17 -> "Applicant Ethnicity",
-      18 -> "Co-applicant Ethnicity",
-      19 -> "Applicant Race: 1",
-      24 -> "Co-applicant Race: 1",
-      29 -> "Applicant Sex",
-      30 -> "Co-applicant Sex",
-      32 -> "Type of Purchaser",
-      37 -> "HOEPA Status",
-      38 -> "Lien Status"
-    )
+  def checkLar(fields: List[String]): ValidationNel[String, List[Int]] = {
 
-    if (values.length != 40 && values.length != 39) {
-      return errors ::: List("Incorrect number of fields: " + values.length)
-    }
+    if (fields.length != 40 && fields.length != 39) {
+      ("Incorrect number of fields: " + fields.length).failure.toValidationNel
```

Comment:
It would also be kind of us to provide the expected number of fields in the error message.  Something like:

    Incorrect number of fields.  found: 38, expected: 39